scripts/valhalla_build_config

```diff
@@ -557,56 +558,67 @@ help_text = {
   }
 }
 
-#used to convert comma separated list of strings into a list
-def comma_separated_list(arg):
-  return arg.split(',')
 
-def parse_bool(arg):
-  if arg.lower() in ('yes', 'true', 't', 'y', '1'):
-    return True
-  elif arg.lower() in ('no', 'false', 'f', 'n', '0'):
-    return False
-  else:
-    raise argparse.ArgumentTypeError('Boolean value expected.')
+def comma_separated_list(arg: str) -> List[str]:
+    """used to convert comma separated list of strings into a list"""
+    return arg.split(',')
 
-#returns a list of leaves of the tree, `\0` separated, stops at non dicts
-#while doing so it also adds arguments to the parser
-def add_leaf_args(path, tree, leaves, parser):
-  #if we are at a dict go deeper
-  if isinstance(tree, dict):
-    for k in tree:
-      v = tree[k]
-      add_leaf_args('\0'.join([path, k]) if len(path) else k, v, leaves, parser)
-  #we've reached a leaf
-  else:
-    keys = path.split('\0')
-    h = help_text
-    for k in keys:
-      h = h[k]
-    t = comma_separated_list if type(tree) == type([]) else (type(tree) if type(tree) != type(optional()) else tree.type)
-    arg = '--' + path.replace('_', '-').replace('\0', '-')
-    parser.add_argument(arg, type = t if t != type(bool()) else parse_bool, help=h, default=tree)
-    leaves.append(path)
 
-#entry point to program
-if __name__ == '__main__':
+def parse_bool(arg: str):
+    if arg.lower() in ('yes', 'true', 't', 'y', '1'):
+        return True
+    elif arg.lower() in ('no', 'false', 'f', 'n', '0'):
+        return False
+    else:
+        raise ArgumentTypeError('Boolean value expected.')
+
+
+def add_leaf_args(path: str, tree: Union[dict, bool, str, list, Optional], leaves_: List[str], parser_: ArgumentParser, help: dict):
```

Comment:
e.g. here I renamed `leaves` to `leaves_`, `parser` to `parser_` to not shadow those outer scope vars